pkg/reconciler/v1alpha1/revision/reconcile_resources.go

```diff
@@ -84,8 +84,8 @@ func (c *Reconciler) reconcileDeployment(ctx context.Context, rev *v1alpha1.Revi
 				if status.Name == resources.UserContainerName {
 					if t := status.LastTerminationState.Terminated; t != nil {
 						rev.Status.MarkContainerExiting(t.ExitCode, t.Message)
+						break
 					}
-					break
```

Comment:
`name` is a primary key, so no two containers can have the same name.  That means the net effect of this change is more iteration when the user-container isn't terminated, but not much else.  I think I prefer the current behavior, which is effectively a search for `user-container` followed by a check on that container.

Reply:
I agree, commented on the original change, I don't think the implementation there is wrong in the first place.